WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorView.swift

```diff
@@ -153,6 +153,11 @@ struct ProductSelectorView: View {
                            viewModel: rowViewModel,
                            onCheckboxSelected: {
                     viewModel.toggleSelectionForAllVariations(of: rowViewModel.productOrVariationID)
+                    // Display the variations list if toggleSelectionForAllVariations is not allowed
+                    if !viewModel.toggleAllVariationsOnSelection {
+                        isShowingVariationList.toggle()
+                        self.variationListViewModel = variationListViewModel
+                    }
```

Comment:
Moving this logic to the view model will be dealt with here: https://github.com/woocommerce/woocommerce-ios/issues/9181